triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/auth/client/AuthServiceFixture.scala

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.engine.trigger.auth.client
+
+import java.net.{InetAddress, ServerSocket, Socket}
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.Uri
+import akka.stream.Materializer
+import com.daml.bazeltools.BazelRunfiles
+import com.daml.ledger.api.domain.LedgerId
+import com.daml.lf.engine.trigger.AuthServiceClient
+import com.daml.platform.common.LedgerIdMode
+import com.daml.platform.sandbox.SandboxServer
+import com.daml.platform.sandbox.config.SandboxConfig
+import com.daml.platform.services.time.TimeProviderType
+import com.daml.ports.Port
+import com.daml.timer.RetryStrategy
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.sys.process.Process
+import scala.concurrent.duration._
+
+object AuthServiceFixture {
+
+  private def findFreePort(): Port = {
+    val socket = new ServerSocket(Port(0).value)
+    try {
+      Port(socket.getLocalPort)
+    } finally {
+      socket.close()
+    }
+  }
+
+  def withAuthServiceClient[A](testName: String)(testFn: AuthServiceClient => Future[A])(
+      implicit system: ActorSystem,
+      mat: Materializer,
+      ec: ExecutionContext): Future[A] = {
+    val adminLedgerId = LedgerId("admin-ledger")
+    val adminLedgerF = for {
+      ledger <- Future(
+        new SandboxServer(
+          SandboxConfig.default.copy(
+            port = Port.Dynamic,
+            timeProviderType = Some(TimeProviderType.Static),
```

Comment:
Any particular reason why you’re using static time instead of wallclock (which is the default on sandbox)?

Reply:
No particular reason, other than copying from the TriggerServiceFixture which does the same. I can remove this here (and probably there as well in another PR).